tmail-backend/imap-extensions/src/test/java/imap/linagora/tmail/imap/TMailMailboxNameConverterTest.java

```diff
@@ -0,0 +1,309 @@
+        void buildFullPathShouldAcceptRelativePathWithSubFolder() {
+            String mailboxName = "mailboxName" + PATH_DELIMITER + "subFolder";
+            assertThat(pathConverter.buildFullPath(mailboxName))
+                .isEqualTo(MailboxPath.forUser(USERNAME_WITH_MAIL, mailboxName));
+        }
+
+        @Test
+        void buildFullPathShouldAcceptAbsoluteUserPathWithSubFolder() {
+            String mailboxName = "mailboxName.subFolder";
+            assertThat(pathConverter.buildFullPath(MailboxConstants.USER_NAMESPACE + PATH_DELIMITER + mailboxName))
+                .isEqualTo(MailboxPath.forUser(USERNAME_WITH_MAIL, mailboxName));
+        }
+
+        @Test
+        void buildFullPathShouldAcceptAbsoluteOtherUserPath() {
+            assertThat(pathConverter.buildFullPath("#user.username2.abc"))
+                .isEqualTo(MailboxPath.forUser(USERNAME2_WITH_MAIL, "abc"));
+        }
+
+        @Test
+        void buildFullPathShouldAcceptAbsoluteOtherUserPathWithSubfolder() {
+            assertThat(pathConverter.buildFullPath("#user.username2.abc.def"))
+                .isEqualTo(MailboxPath.forUser(USERNAME2_WITH_MAIL, "abc.def"));
+        }
+
+        @Test
+        void mailboxNameShouldReturnNameOnlyWhenRelativeAndUserMailbox() {
+            assertThat(pathConverter.mailboxName(RELATIVE, MailboxPath.forUser(USERNAME_WITH_MAIL, "abc"), mailboxSession))
+                .isEqualTo("abc");
+        }
+
+        @Test
+        void mailboxNameShouldReturnFQDNWhenRelativeAndOtherUserMailbox() {
+            assertThat(pathConverter.mailboxName(RELATIVE, MailboxPath.forUser(USERNAME2_WITH_MAIL, "abc"), mailboxSession))
+                .isEqualTo("#user.username2.abc");
+        }
+
+        @Test
+        void mailboxNameShouldReturnFQDNWhenRelativeAndSharedMailbox() {
+            assertThat(pathConverter.mailboxName(RELATIVE, new MailboxPath("#Shared", Username.of("marketing@apache.org"), "abc"), mailboxSession))
+                .isEqualTo("#Shared.marketing.abc");
+        }
+
+        @Test
+        void mailboxNameShouldReturnFQDNWhenNotRelativeAndUserMailbox() {
+            assertThat(pathConverter.mailboxName(!RELATIVE, MailboxPath.forUser(USERNAME_WITH_MAIL, "abc"), mailboxSession))
+                .isEqualTo("#private.abc");
+        }
+
+        @Test
+        void mailboxNameShouldReturnFQDNWhenNotRelativeAndOtherUserMailbox() {
+            assertThat(pathConverter.mailboxName(!RELATIVE, MailboxPath.forUser(USERNAME2_WITH_MAIL, "abc"), mailboxSession))
+                .isEqualTo("#user.username2.abc");
+        }
+
+        @Test
+        void mailboxNameShouldReturnFQDNWhenNotRelativeAndSharedMailbox() {
+            assertThat(pathConverter.mailboxName(!RELATIVE, new MailboxPath("#Shared", Username.of("marketing@apache.org"), "abc"), mailboxSession))
+                .isEqualTo("#Shared.marketing.abc");
+        }
+    }
+    /*
+        Copied from org.apache.james.imap.main.PathConverterTest
```

Comment:
This copy is a shame.

How about writing on James side a `PathConverterBasicContract` that PathConverterTest and `TMailMailboxNameConverterTest` could reuse?